### PR TITLE
[DA-1911] Change tracking and other modifications

### PR DESCRIPTION
**Pull request by kenny-skaggs**
This updates the data-dictionary code to track changes made to the data-dictionary spreadsheet and incorporate other feedback from Alpha
- Deprecation notes are made with the RDR version number that they were deprecated in
- Any new rows added to the dictionary show what RDR version they were added in
- Refactored the column indexes for the dictionary schema tabs into an Enum, and created the DictionarySchemaRowUpdateHelper to help write schema rows to the sheet (to help with writing to the change log)
- Created the KeyTabUpdateHelper to help with the change indicators for the key tabs too
- Added functionality for tracking changes (will give specific changes, what rows were created/removed or how they were modified, for the schema tabs, and will give a boolean for whether the key tabs were changed)
- Showing an indicator for PPI questionnaires on the questionnaire key tab
- Showing organization information on the site key tab

One more PR for this ticket is on the way. The next PR will tie it together with the deploy process to update the data-dictionary each time we deploy to Prod.

### CODE REVIEW

File: rdr_service/services/data_dictionary_updater.py [hunk 1]
```diff
@@ -179,48 +323,67 @@ def _get_column_should_show_unique_values(self, column_definition, analyzer: Mod
         return False
 
     def _populate_questionnaire_key_tab(self, sheet: GoogleSheetsClient):
-        query = self.session.query(QuestionnaireHistory).options(joinedload(QuestionnaireHistory.concepts))
-        questionnaire_data_list = query.all()
+        update_helper = KeyTabUpdateHelper()
```
Comment (dtharpeuno) [inline rdr_service/services/data_dictionary_updater.py:326]: Can you invoke this line, the same way you are doing the above args for sheet?

File: rdr_service/services/data_dictionary_updater.py [hunk 2]
```diff
@@ -179,48 +323,67 @@ def _get_column_should_show_unique_values(self, column_definition, analyzer: Mod
...
-            sheet.update_cell(row_number, 3, 'Y' if has_response else 'N')
+            update_helper.update_with_values(sheet, row_number, [
+                questionnaire_id, code_display, code_short_value, has_responses_yn_indicator, is_ppi_survey_yn_indicator
+            ])
+
+        sheet.truncate_tab_at_row(len(questionnaire_data_list))
+        self.changelog[questionnaire_key_tab_id] = update_helper.change_detected
 
     def _populate_hpo_key_tab(self, sheet: GoogleSheetsClient):
+        update_helper = KeyTabUpdateHelper()
```
Comment (dtharpeuno) [inline rdr_service/services/data_dictionary_updater.py:360]: Same as comment above.

**Review by dtharpeuno — APPROVED** [review]
Have to share some secrets on the workflow for building this!!!!
Looks awesome.

File: rdr_service/services/data_dictionary_updater.py (hunk 1 above)
Comment (kenny-skaggs) [reply rdr_service/services/data_dictionary_updater.py:326]: Resolved offline

**Pull request merged**